Parca crashloop

Hey folks, I deployed parca v0.2.0 release and get the following crashloop error:

```
➜ kubectl logs -n parca parca-6df694b585-cb29p
ooooooooo.
`888   `Y88.
 888   .d88'  .oooo.   oooo d8b  .ooooo.   .oooo.
 888ooo88P'  `P  )88b  `888""8P d88' `"Y8 `P  )88b
 888          .oP"888   888     888        .oP"888
 888         d8(  888   888     888   .o8 d8(  888
o888o        `Y888""8o d888b    `Y8bod8P' `Y888""8o



name=parca ts=2021-10-21T11:20:27.271462421Z caller=log.go:168 level=info msg="loading bucket configuration"
level=info name=parca ts=2021-10-21T11:20:27.271832107Z caller=server.go:85 msg="starting server" addr=:7070
level=error name=parca ts=2021-10-21T11:20:47.518687354Z caller=symbol.go:132 msg="failed to open object file" file=tmp/334830476d546a6656644163694f7a752d3662362f36474e7a6d325156336f6552756f7a58765879592f42567078344e346d56647a704361795a307637572f76636e7a51556f364e717568466554313178536c/debuginfo start=4194304 limit=22261760 offset=0 err="failed to read DWARF data: unexpected EOF"
panic: sync/atomic: store of nil value into Value

goroutine 89 [running]:
sync/atomic.(*Value).Store(0xc003180030, 0x0, 0x0)
	sync/atomic/value.go:47 +0x12c
github.com/goburrow/cache.(*entry).setValue(...)
	github.com/goburrow/cache@v0.1.4/policy.go:59
github.com/goburrow/cache.newEntry(...)
	github.com/goburrow/cache@v0.1.4/policy.go:50
github.com/goburrow/cache.(*localCache).Put(0xc0001cb600, 0x141c680, 0xc00266a060, 0x0, 0x0)
	github.com/goburrow/cache@v0.1.4/local.go:124 +0x216
github.com/parca-dev/parca/pkg/symbol.(*Symbolizer).NewLiner(0xc00007d3b0, 0xc0039aa2a0, 0xc0003a43c0, 0xb4, 0xa6, 0xc0003a43c0, 0xb4, 0x0)
	github.com/parca-dev/parca/pkg/symbol/symbol.go:111 +0x542
github.com/parca-dev/parca/pkg/debuginfo.(*Store).Symbolize(0xc00003ae00, 0x198ea08, 0xc00003ae80, 0xc0039aa2a0, 0xc003596680, 0xb, 0x10, 0x500, 0x997e65, 0x22fef20)
	github.com/parca-dev/parca/pkg/debuginfo/store.go:205 +0x376
github.com/parca-dev/parca/pkg/symbolizer.(*Symbolizer).symbolize(0xc0001bced0, 0x198ea08, 0xc00003ae80, 0xc00433c000, 0x8f0, 0x900, 0x0, 0x0)
	github.com/parca-dev/parca/pkg/symbolizer/symbolizer.go:87 +0x7ef
github.com/parca-dev/parca/pkg/symbolizer.(*Symbolizer).Run.func1(0xc000098ec0, 0xc000275eb0)
	github.com/parca-dev/parca/pkg/symbolizer/symbolizer.go:57 +0xb1
github.com/parca-dev/parca/pkg/runutil.Repeat(0x2540be400, 0xc0000f6000, 0xc002df5f28, 0x0, 0x0)
	github.com/parca-dev/parca/pkg/runutil/runutil.go:38 +0x88
github.com/parca-dev/parca/pkg/symbolizer.(*Symbolizer).Run(0xc0001bced0, 0x198ea08, 0xc00003ae80, 0x2540be400, 0x0, 0x0)
	github.com/parca-dev/parca/pkg/symbolizer/symbolizer.go:47 +0x8e
github.com/parca-dev/parca/pkg/parca.Run.func1(0x0, 0x0)
	github.com/parca-dev/parca/pkg/parca/parca.go:174 +0x4b
github.com/oklog/run.(*Group).Run.func1(0xc0000bd020, 0xc0000c5220, 0xc0001bcf00)
	github.com/oklog/run@v1.1.0/group.go:38 +0x27
created by github.com/oklog/run.(*Group).Run
	github.com/oklog/run@v1.1.0/group.go:37 +0xbb
```

This is deployed on OpenShift cluster, the previous version worked correctly.

Same error happens on my Kubernetes cluster, this is scaleaway cluster with v1.22 kube version:

```
kubectl logs -n parca parca-6dbfb6d457-kxfxf -p
ooooooooo.
`888   `Y88.
 888   .d88'  .oooo.   oooo d8b  .ooooo.   .oooo.
 888ooo88P'  `P  )88b  `888""8P d88' `"Y8 `P  )88b
 888          .oP"888   888     888        .oP"888
 888         d8(  888   888     888   .o8 d8(  888
o888o        `Y888""8o d888b    `Y8bod8P' `Y888""8o



name=parca ts=2021-10-21T11:27:30.541368126Z caller=log.go:168 level=info msg="loading bucket configuration"
level=info name=parca ts=2021-10-21T11:27:30.541750458Z caller=server.go:85 msg="starting server" addr=:7070
level=error name=parca ts=2021-10-21T11:28:20.712754483Z caller=symbol.go:132 msg="failed to open object file" file=tmp/59346a657852657948716658516133766d314c742f457976554561646a48664a466466787936634a422f6f47686a694a576935783850304c41794b3138552f4169636649475a584b3541684452796463667a66/debuginfo start=4194304 limit=15298560 offset=0 err="failed to read DWARF data: unexpected EOF"
panic: sync/atomic: store of nil value into Value

goroutine 92 [running]:
sync/atomic.(*Value).Store(0xc003e80030, 0x0, 0x0)
	sync/atomic/value.go:47 +0x12c
github.com/goburrow/cache.(*entry).setValue(...)
	github.com/goburrow/cache@v0.1.4/policy.go:59
github.com/goburrow/cache.newEntry(...)
	github.com/goburrow/cache@v0.1.4/policy.go:50
github.com/goburrow/cache.(*localCache).Put(0xc00029ef20, 0x141c680, 0xc0000a4110, 0x0, 0x0)
	github.com/goburrow/cache@v0.1.4/local.go:124 +0x216
github.com/parca-dev/parca/pkg/symbol.(*Symbolizer).NewLiner(0xc000296fa0, 0xc003e20ae0, 0xc003934540, 0xb4, 0xa6, 0xc003934540, 0xb4, 0x0)
	github.com/parca-dev/parca/pkg/symbol/symbol.go:111 +0x542
github.com/parca-dev/parca/pkg/debuginfo.(*Store).Symbolize(0xc0005d9e00, 0x198ea08, 0xc0005d9e80, 0xc003e20ae0, 0xc003bb9000, 0xa4, 0x100, 0x80, 0x997e65, 0x22fef20)
	github.com/parca-dev/parca/pkg/debuginfo/store.go:205 +0x376
github.com/parca-dev/parca/pkg/symbolizer.(*Symbolizer).symbolize(0xc00002d140, 0x198ea08, 0xc0005d9e80, 0xc003da6000, 0x355, 0x400, 0x0, 0x0)
	github.com/parca-dev/parca/pkg/symbolizer/symbolizer.go:87 +0x7ef
github.com/parca-dev/parca/pkg/symbolizer.(*Symbolizer).Run.func1(0xc00264cec0, 0xc00033deb0)
	github.com/parca-dev/parca/pkg/symbolizer/symbolizer.go:57 +0xb1
github.com/parca-dev/parca/pkg/runutil.Repeat(0x2540be400, 0xc0000c8000, 0xc003b99f28, 0x0, 0x0)
	github.com/parca-dev/parca/pkg/runutil/runutil.go:38 +0x88
github.com/parca-dev/parca/pkg/symbolizer.(*Symbolizer).Run(0xc00002d140, 0x198ea08, 0xc0005d9e80, 0x2540be400, 0x0, 0x0)
	github.com/parca-dev/parca/pkg/symbolizer/symbolizer.go:47 +0x8e
github.com/parca-dev/parca/pkg/parca.Run.func1(0x0, 0x0)
	github.com/parca-dev/parca/pkg/parca/parca.go:174 +0x4b
github.com/oklog/run.(*Group).Run.func1(0xc000029200, 0xc0004dd600, 0xc00002d170)
	github.com/oklog/run@v1.1.0/group.go:38 +0x27
created by github.com/oklog/run.(*Group).Run
	github.com/oklog/run@v1.1.0/group.go:37 +0xbb
```

--- a/pkg/symbol/symbol.go
+++ b/pkg/symbol/symbol.go
@@ -83,34 +83,34 @@ func NewSymbolizer(logger log.Logger, opts ...Option) (*Symbolizer, error) {
 	return sym, nil
 }
 
-func (s *Symbolizer) NewLiner(m *profile.Mapping, path string) (lnr liner, err error) {
-	hash, err := hash(path)
+func (s *Symbolizer) NewLiner(m *profile.Mapping, path string) (liner, error) {
+	h, err := hash(path)
 	if err != nil {
 		level.Warn(s.logger).Log("msg", "failed to generate cache key", "err", err)
-		hash = path
+		h = path
 	}
 
 	// Check if we already attempt to build a liner for this path.
-	if _, failedBefore := s.failed[hash]; failedBefore {
+	if _, failedBefore := s.failed[h]; failedBefore {
 		level.Debug(s.logger).Log("msg", "already failed to create liner for this debug info file, skipping")
 		return nil, ErrLinerFailedBefore
 	}
 
-	if val, ok := s.cache.GetIfPresent(hash); ok {
+	if val, ok := s.cache.GetIfPresent(h); ok {
 		level.Debug(s.logger).Log("msg", "using cached liner to resolve symbols", "file", path)
 		return val.(liner), nil
 	}
 
-	lnr, err = s.newLiner(m, path)
+	lnr, err := s.newLiner(m, path)
 	if err != nil {
-		s.failed[hash] = struct{}{}
+		s.failed[h] = struct{}{}
 		s.cache.Invalidate(hash)
-		return
+		return nil, err
 	}
 
 	level.Debug(s.logger).Log("msg", "liner cached", "file", path)
-	s.cache.Put(hash, lnr)
-	return
+	s.cache.Put(h, lnr)
+	return lnr, nil
 }
 
 func (s *Symbolizer) Close() error {
@@ -169,20 +169,7 @@ func (s *Symbolizer) newLiner(m *profile.Mapping, path string) (liner, error) {
 		)
 	}
 
-	// Just in case, underlying DWARF can symbolize addresses.
-	level.Debug(s.logger).Log("msg", "falling back to DWARF liner resolve symbols", "file", path)
-	lnr, err := addr2line.DWARF(s.logger, s.demangler, s.attemptThreshold, m, path)
-	if err != nil {
-		level.Error(s.logger).Log("msg", "failed to open object file",
-			"file", path,
-			"start", m.Start,
-			"limit", m.Limit,
-			"offset", m.Offset,
-			"err", err,
-		)
-		return nil, err
-	}
-	return lnr, nil
+	return nil, errors.New("cannot create a liner from given object file")
 }
 
 func hash(path string) (string, error) {

